Stock.Api/Controllers/ProviderController.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Stock.Api.DTOs;
+using Stock.Api.Extensions;
+using Stock.AppService.Services;
+using Stock.Model.Entities;
+
+namespace Stock.Api.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/provider")]
+    [ApiController]
+    public class ProviderController: ControllerBase
+    {
+        private readonly ProviderService service;
+        private readonly IMapper mapper;
+        
+        public ProviderController(ProviderService service, IMapper mapper)
+        {
+            this.service = service;
+            this.mapper = mapper;
+        }
+            /// <summary>
+            /// Permite recuperar todas las instancias
+            /// </summary>
+            /// <returns>Una colección de instancias</returns>
+            [HttpGet]
+
+            public ActionResult<IEnumerable<ProviderDTO>> Get()
+            {
+                return this.mapper.Map<IEnumerable<ProviderDTO>>(this.service.GetAll()).ToList();
+            }
+
+            /// <summary>
+            /// Permite crear una nueva instancia
+            /// </summary>
+            /// <param name="value">Una instancia</param>
+            [HttpPost]
+            public Provider Post([FromBody] ProviderDTO value)
+            {
+                TryValidateModel(value);
+                var Provider = this.service.Create(this.mapper.Map<Provider>(value));
+                return this.mapper.Map<Provider>(Provider);
```

Comment:
Estaría bueno que el código que llama a la lógica del controlador esté inserto en uno o varios try and catch para evitar que un error rompa la app y a la vez poder manejar de manera eficiente las excepciones y brindar una respuesta al cliente dependiendo de ellas.  Si revisas el StoreController veras algunos ejemplos.

Reply:
Esto fue corregido con el nuevo commit!